x/evidence/keeper/grpc_query.go

```diff
@@ -0,0 +1,94 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	"github.com/cosmos/cosmos-sdk/types/query"
+
+	proto "github.com/gogo/protobuf/proto"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/evidence/exported"
+	"github.com/cosmos/cosmos-sdk/x/evidence/types"
+)
+
+var _ types.QueryServer = Keeper{}
+
+// Evidence implements the Query/Evidence gRPC method
+func (k Keeper) Evidence(c context.Context, req *types.QueryEvidenceRequest) (*types.QueryEvidenceResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	if req.EvidenceHash == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid hash")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	evidence, _ := k.GetEvidence(ctx, req.EvidenceHash)
+	if evidence == nil {
+		return nil, status.Errorf(codes.NotFound, "evidence %s not found", req.EvidenceHash)
+	}
+
+	evidenceAny, err := ConvertEvidence(evidence)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
```

Comment:
can you add a test case with an unregistered evidence type?

Reply:
do you mean with some other type than `Equivocation`? How could it have been submitted in the first place?

Reply:
yeah or another protobuf Any message type that is not an Evidence

Reply:
@fedekunze this is probably impossible to test. I've given @blushi some guidance. Looking at the surrounding code (`GetEvidence`) there's actually no way to trigger this condition. We just need to add `proto.Message` to the `Evidence` interface which is needed anyway and `ConvertEvidence` can go away.

Reply:
Regarding @aaronc's suggestion, we unfortunately can't just add `proto.Message` to the `Evidence` interface at this point because `Evidence struct` in `x/ibc/07-tendermint` doesn't implement it. 
In the context of the test, it's just impossible to make that particular test case fail because an `Evidence` can't even be submitted (and then returned by `GetEvidence` in the query) if it doesn't implement `proto.Message` (see https://github.com/cosmos/cosmos-sdk/blob/marie/5921-grpc-x-evidence/x/evidence/keeper/keeper.go#L105-L108) Trying to submit it would cause the program to panic. With regards to that, should we resolve this?